### ISSUE TITLE
URL's with encoded characters such as white space (`%20`) cause mdformat to fail unexpectedly.

### ISSUE THREAD

**Issue by butler54** [executablebooks/mdformat#254]
### Describe the problem

# Problem
Various URLs are frequently captured in markdown. While in ideal world they would all be wrapped in a link object `[name of link](https://myurl.com)`. Unfortunately it is common that URLs are dumped in a document directly.

The result is we may have URL's where white space is in the url. e.g.
```markdown
https://mytest.com/files/word%$20document.docx
```
which renders when expanded to `https://mytest.com/files/word document.docx`.

The result is mdformat fails as the rendered html is different to the source format.
```

Error: Could not format
"/path/to/markdown.md".

The formatted Markdown renders to different HTML than the input Markdown. This
is likely a bug in mdformat. Please create an issue report here, including the
input Markdown: https://github.com/executablebooks/mdformat/issues
```

Note: Github (and some other renderers) do not render out the URLS.
Note: This happens generally, however, URL's are the problem case which needs to be dealt with.

Technically from a gfm flavoured markdown perspective you could insist URL's like this are [encoded with autolinks](https://github.github.com/gfm/#autolink), however, today this also fails due to the %20 characters.

# Potential solution
- The challenge here is the error handling at the moment is challenging to figure out 'what' is wrong. A scan for percentage encoded characters would be a good start so an alert could be thrown
- For mdformat-gfm support for autolinks as a minimum would help
or:
- Examining the rendering solution being used which is causing this issue.

### Steps to reproduce

Create a md document.

Create a URL (or any object) containing a %20 or similar encoded piece of text.

Attempt to run mdformat (with mdformat-gfm enabled).

### The version of Python you're using

3.8

### Your operating system

Mac os

### Versions of your packages

mdformat 0.7.8

**Comment by hukkin** [comment]
Thanks for the issue!

A note to self: this issue only concerns [the GFM plugin](https://github.com/hukkin/mdformat-gfm) and does not reproduce on a clean mdformat installation with no plugins.

**Comment by hukkin** [comment]
This should be fixed in `mdformat>=0.7.9` and `mdformat-gfm>=0.3.3`
